examples/common/maths.adb

```diff
@@ -206,6 +204,45 @@ package body Maths is
     end Scaling_Matrix;
 
     --  ------------------------------------------------------------------------
+
+    function To_3 (Array_4 : Single_Array) return Single_Array is
+        Array_3  : Single_Array (0 .. Array_4'Length * 3 / 4 - 1);
```

Comment:
Ada arrays start at `1` by convention; your code would be simpler if you adhere to that (you can drop the `- 1`).